libs/hdf-converters/src/msft-secure-score-mapper.ts

```diff
@@ -0,0 +1,202 @@
+              transformer: (d: ControlScore) => d.description || ''
+            },
+            impact: {
+              transformer: (d: ControlScore) => {
+                // return controlCategory from the profile document where its id matches the controlName
+                const knownMaxScores = this.getProfiles(
+                  d.controlName || ''
+                ).map((p) => p.maxScore || 0);
+
+                if (knownMaxScores.length === 0) {
+                  return 0.5;
+                }
+
+                const highMaxScore = Math.max(...knownMaxScores);
+                return highMaxScore / 10.0;
+              }
+            },
+            refs: [],
+            tags: {
+              group: {
+                transformer: (d: ControlScore) => {
+                  // return controlCategory from the profile document where its id matches the controlName
+                  return this.getProfiles(d.controlName || '').map(
+                    (p) => p.controlCategory
+                  );
+                }
+              },
+              tiers: {
+                transformer: (d: ControlScore) => {
+                  // return tiers from the profile document where its id matches the controlName
+                  return this.getProfiles(d.controlName || '').map(
+                    (p) => p.tier
+                  );
+                }
+              },
+              threats: {
+                transformer: (d: ControlScore) => {
+                  // return unique threats from the profile document where its id matches the controlName
+                  const uniqs: Set<string> = new Set();
+                  this.getProfiles(d.controlName || '').forEach((p) =>
+                    p.threats?.forEach((threat) => uniqs.add(threat))
+                  );
+                  return [...uniqs];
+                }
+              },
+              services: {
+                transformer: (d: ControlScore) => {
+                  // return thrserviceeats from the profile document where its id matches the controlName
+                  return this.getProfiles(d.controlName || '').map(
+                    (p) => p.service
+                  );
+                }
+              },
+              userImpacts: {
+                transformer: (d: ControlScore) => {
+                  // return userImpacts from the profile document where its id matches the controlName
+                  return this.getProfiles(d.controlName || '')
+                    .filter((p) => p.userImpact !== undefined)
+                    .map((p) => p.userImpact);
+                }
+              }
+            },
+            source_location: {},
+            code: {
+              transformer: (
+                d: ControlScore & {implementationStatus: string}
+              ) => {
+                const profiles = this.getProfiles(d.controlName || '');
+
+                if (profiles?.length === 0) {
+                  return d.implementationStatus;
```

Comment:
We're returning this if there are no profiles matching the control name? What if there are?

Reply:
just pushed a better implementation.

always include implementationStatus from control.  include from profiles if available.